stan/math/fwd/fun/tcrossprod.hpp

```diff
@@ -18,7 +18,7 @@ tcrossprod(const EigMat& m) {
     return {};
   }
   const auto& m_ref = to_ref(m);
-  return multiply(m_ref, m_ref.transpose());
+  return m_ref * m_ref.transpose();
```

Comment:
I think we could probably just delete the forward implementation of this now.

Multiply returns a scalar if you pass a row vector into this which does not match the prim/reverse mode behavior.

When I added Vector/RowVector tests to tcrossprod this popped up (and we need Vector/RowVector support in tcrossprod cause of how it is used elsewhere in the library)

Reply:
I'm fine with that, idt this is doing much of anything special